src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1035,6 +1035,7 @@ static bool patternMatchesNull(BoundPattern pattern)
                     case BoundITuplePattern:
                     case BoundRelationalPattern:
                     case BoundDeclarationPattern:
+                    case BoundListPattern:
```

Comment:
There's another switch with a missing case in patternMatchesNull above. 

It's not clear how an exhaustive switch is supposed to help here since not every pattern node is going through this check as demonstrated in this bug. Unless there's a compile-time diagnostic, you'd need to scan the code to find each instance or else, wait for a crash later. Would it make sense to have a safe default here instead of throwing?

Reply:
Thanks. Fixed that other case, plus two more (for slices).

I added an assertion that should help catch this problem with future patterns.